### ISSUE TITLE
Strange: Raw value is OK but value has added 1cbm

### ISSUE THREAD

**Issue by hjgode** [jomjol/AI-on-the-edge-device#763]
Hello

I have a strange issue, I set the pre Value to 350. something and after some time, the raw value is OK but the value (transmitted to MQTT) show 1 or 2 cbm more: (I know the images can be sharper but that should no be the issue with the logic)

![incorrect_reading_01](https://user-images.githubusercontent.com/3365637/162386960-6d84290a-8e0f-4162-b8de-e0c91cd6dced.png)

 and another day:

![fehler_raw_OK](https://user-images.githubusercontent.com/3365637/162387065-ecf7fee9-8711-4a4e-ba1a-9799adbca2e1.png)

my config.ini:

`[MakeImage]
;LogImageLocation = /log/source
WaitBeforeTakingPicture = 5
;LogfileRetentionInDays = 15
Brightness = 0
Contrast = 0
Saturation = 0
LEDIntensity = 20
ImageQuality = 5
ImageSize = VGA
FixedExposure = true

[Alignment]
InitialRotate = 1
InitialMirror = false
SearchFieldX = 20
SearchFieldY = 20
AlignmentAlgo = fast
FlipImageSize = false
/config/ref0.jpg 215 167
/config/ref1.jpg 508 152

[Digits]
Model = /config/dig-s2-q-20220211.tflite
;LogImageLocation = /log/digit
;LogfileRetentionInDays = 3
ModelInputSize = 20 32
main.dig3 376 138 28 52
main.dig4 421 145 29 49
main.dig5 464 143 30 48

[Analog]
Model = /config/ana-s2-q-20220213.tflite
;LogImageLocation = /log/analog
;LogfileRetentionInDays = 3
ModelInputSize = 32 32
main.n1 509 235 88 90
main.n01 450 330 93 95
main.n001 373 389 57 53
main.n0001 242 324 82 83

[PostProcessing]
main.DecimalShift = 0
PreValueUse = true
PreValueAgeStartup = 720
AllowNegativeRates = false
main.MaxRateValue = 1
main.MaxRateType = AbsoluteChange
;main.ExtendedResolution = false
main.IgnoreLeadingNaN = true
ErrorMessage = false
CheckDigitIncreaseConsistency = true

[MQTT]
Uri = mqtt://192.168.0.40:1883
MainTopic = wasserzaehler
;ClientID = wasser
;user = USERNAME
;password = PASSWORD

;[GPIO]
;IO0 = input disabled 10 false false 
;IO1 = input disabled 10 false false 
;IO3 = input disabled 10 false false 
;IO4 = built-in-led disabled 10 false false 
;IO12 = input-pullup disabled 10 false false 
;IO13 = input-pullup disabled 10 false false 
LEDType = WS2812
LEDNumbers = 2
LEDColor = 50 50 50

[AutoTimer]
AutoStart = true
Intervall = 5

[Debug]
Logfile = false
LogfileRetentionInDays = 3

[System]
TimeZone = CET-1CEST,M3.5.0,M10.5.0/3
;TimeServer = undefined
;AutoAdjustSummertime = false
;Hostname = undefined
SetupMode = false
`

**Comment by stonemaan** [comment]
I have pretty much the same error. I enabled allow negative rates. And it does not decrease the value over time. Recognition of the raw works perfectly, but it brings the wrong value.

latest Firmware installed.

After Setting pre value, over time, it does Switch to a higher value without any reason.
![8CFDC68D-4475-41A4-BDC5-38B94FF78198](https://user-images.githubusercontent.com/73359164/162478171-591e64e7-5ec2-4151-b1ba-20ffd33d9452.jpeg)

**Comment by jomjol** [comment]
You have enabled `CheckDigitIncreaseConsistency` and in a previous round there was a zero crossing in the subdigit detected (e.g. 359.**6** --> 359.**5**). This triggers the `CheckDigitIncreaseConsistency` because it assumes, that there was a zero crossing and therefore it results in 360.5
This is a not perfect reading of the 0.1x pointer. You can disable the `CheckDigitIncreaseConsistency` or make the `MaxRateValue` smaller (1 is really big!) to reject such a big change.

**Comment by stonemaan** [comment]
Disableling CheckDigitIncreaseConsistency worked. Thanks for your help. I was not aware of that.

**Comment by hjgode** [comment]
Hi jomjol

the config was not provided by stonemaan, but the hint with CheckDigitIncreaseConsistency  helped him. OK.

Back to my issue, why does the raw value show the correct value of 350.6541 and the transmitted value show 351.6541?

What is the MaxRateValue for? I think it is about the maximum change value of the whole value. As we are looking at cubic meters with a resolution of 4 decimal places, I assume the MaxRateValue=1 means a max change of 1 cbm (or 1.0000). So the logic would allow a change from 350.6541 to 351.6541 but not from 350.6541 to 360,6541.

In the mean time I changed the focus and rotation of the 'sensor'. I had to apply 89° pre rotation value and came up with scattered images in the main screen. I had to also set FlipImage resolution, to get a clean image in the main screen (and a workable recognition).

![withrotation_and_flipres](https://user-images.githubusercontent.com/3365637/162558331-c4b41cf0-2a07-4e59-8838-6c47bd4e0db9.png)
